### ISSUE TITLE
Planning to have custom swiper with other features like only including lazy loading ? 

### ISSUE THREAD

**Issue by intelcoder** [kidjp85/react-id-swiper#216]
Currently, I only need lazy loading feature but I still have to load all of the other feature. Is there any plan to optionally select those feature for the custom swiper ?

**Comment by CarlosOlave** [comment]
+1 on this question.

**Comment by kidjp85** [comment]
For now I don't think it can be supported. Even for custom version, I have to customise Swiper team config and build manually.

**Comment by Ephys** [comment]
@kidjp85 

I think there is a way to do it:

You could import the following module:
`import { Swiper } from 'swiper/dist/js/swiper.esm.js';`
instead of
`import Swiper from 'swiper/dist/js/swiper';`

and require the user to pass the different modules they need through props:

```
import Swiper from 'react-id-swiper';
import { Navigation, Pagination, Scrollbar } from 'swiper/dist/js/swiper.esm.js';

function MyPrettySlider() {
  return (
    <Swiper enableModules={[Navigation, Pagination, Scrollbar]}>
      <div>Slide 1</div>
      <div>Slide 2</div>
      <div>Slide 3</div>
      <div>Slide 4</div>
      <div>Slide 5</div>
    </Swiper>
  );
}
```

then react-id-swiper can call `Swiper.use(this.props.enableModules);` at the beginning of `buildSwiper`

---

This could even replace your current implementation of `custom` as by default it wouldn't have any module enabled (so, lightweight) and it would allow us to enable them if we so choose.

**Comment by abstractmatter** [comment]
+1 on this solution!

**Comment by kidjp85** [comment]
@Ephys : That will be a great approach if it happens. My concern now is we may have trouble with typescript when importing `swiper/dist/js/swiper.esm.js`

**Comment by Ephys** [comment]
Good point!

Looked into it and `swiper/dist/js/swiper.esm.js` [looks supported](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/swiper/dist/js/swiper.esm.d.ts) by DefinitelyTyped 
Users would need to install [`@types/swiper`](https://www.npmjs.com/package/@types/swiper) of course

**Comment by kidjp85** [comment]
@Ephys . Now it's available on `2.1.0`. Can be tested on prepared Codesanbox.

**Comment by Ephys** [comment]
Oh wow that was fast @kidjp85, thank you very much we'll check it out! :)
